src/apps/api/serializers/submissions.py

```diff
@@ -142,11 +143,13 @@ def validate(self, attrs):
             fact_sheet = data['phase'].competition.fact_sheet
             if set(fact_sheet_answers.keys()) != set(fact_sheet.keys()):
                 raise ValidationError("Fact Sheet keys do not match Answer keys")
-            for key in fact_sheet_answers.keys():
+            for key in fact_sheet_answers:
```

Comment:
you could do..
```python
for key, value in fact_sheet_answers.items():
```

then you don't have to dereference `fact_sheet_answers[key]` over and over, can use `value`